### ISSUE TITLE
Document XXX Comment Policy

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2909]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#2909|
|Type|enhancement|
|Created|2007-11-19 03:38:02Z|
|Branch|https://github.com/twisted/twisted/tree/xxx-comment-policy-2909-2|

Add the following to the coding standard:

XXX (or TODO) comments must include a date, the name of the person who filed it, a reference to the associated bug and some description of the conditions under which the commented code can be corrected and the comment removed.

<details><summary>Searchable metadata</summary>

```
trac-id__2909 2909
type__enhancement enhancement
reporter__jml jml
priority__normal normal
milestone__ 
branch__branches_xxx_comment_policy_2909_2 branches/xxx-comment-policy-2909-2
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1195443482000000 1195443482000000
changetime__1221547288000000 1221547288000000
version__None None
owner__ 
cc__thijs cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [24693]) Branching to 'xxx-comment-policy-2909'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

I think a lot of this information is redundant.  Such comments should include a ticket number.  All the other information is present on the ticket.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra set status to `assigned`|
|-|-|

Replying to [exarkun](#note_4):
> I think a lot of this information is redundant.  Such comments should include a ticket number.  All the other information is present on the ticket.

I'll remove that redundant info. Any other coding standards for comments?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra set owner to @exarkun<br>@thijstriemstra set status to `new`|
|-|-|

See r24837. I couldn't find another paragraph where this note about comments made sense, so shortened the sentence and kept the dedicated section for comments.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

Text looks good.  There's a simple conflict when merging the branch to trunk now, should be obvious how to resolve it.  Please do so with a -2 branch and then merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra commented|
|-|-|

(In [24861]) Branching to 'xxx-comment-policy-2909-2'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [24863]) Merge xxx-comment-policy-2909-2: Document comment policy in coding standard.

Author: jml, thijs
Reviewer: exarkun
Fixes: [#2909](https://github.com/twisted/twisted/issues/2909)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
